azurerm/resource_arm_key_vault.go

```diff
@@ -314,8 +338,21 @@ func resourceArmKeyVaultRead(d *schema.ResourceData, meta interface{}) error {
 		d.Set("enabled_for_template_deployment", props.EnabledForTemplateDeployment)
 		d.Set("vault_uri", props.VaultURI)
 
-		if err := d.Set("sku", flattenKeyVaultSku(props.Sku)); err != nil {
-			return fmt.Errorf("Error setting `sku` for KeyVault %q: %+v", *resp.Name, err)
+		if skuName == "" {
```

Comment:
Same as above, we can vastly simplify this./

Reply:
Yep